metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -174,81 +178,110 @@ object SbtBuildTool {
-        Files.write(metalsPluginFile.toNIO, bytes)
-      }
+    val bytes =
+      sbtPlugin(plugin).getBytes(StandardCharsets.UTF_8)
+
+    projectDir.toFile.mkdirs()
+    val metalsPluginFile = projectDir.resolve("metals.sbt")
+    val pluginFileShouldChange = !metalsPluginFile.isFile ||
+      !metalsPluginFile.readAllBytes.sameElements(bytes)
+
+    if (pluginFileShouldChange) {
+      Files.write(metalsPluginFile.toNIO, bytes)
     }
   }
 
+  /**
+   * Write all the plugins used by Metals when connected to sbt server:
+   * - the sbt-metals plugin in the project directory
+   * - the sbt-jdi-tools plugin in the project/project directory
+   */
+  def writeSbtMetalsPlugins(
+      workspace: AbsolutePath
+  ): Unit = {
+    val mainMeta = workspace.resolve("project")
+    val metaMeta = workspace.resolve("project").resolve("project")
+    writeSinglePlugin(mainMeta, metalsPluginDetails)
+    writeSinglePlugin(metaMeta, jdiToolsPluginDetails)
+  }
+
+  case class PluginDetails private (
+      description: Seq[String],
+      artifact: String,
+      resolver: Option[String]
+  )
+
   /**
    * Short description and artifact for the sbt-bloop plugin
    */
-  private def bloopPluginDetails(version: String) =
-    (
-      "This file enables sbt-bloop to create bloop config files.",
-      s""""ch.epfl.scala" % "sbt-bloop" % "$version""""
+  def bloopPluginDetails(version: String): PluginDetails = {
+    val resolver =
+      if (isSnapshotVersion(version))
+        Some("""Resolver.bintrayRepo("scalacenter", "releases")""")
+      else None
+
+    PluginDetails(
+      description =
+        Seq("This file enables sbt-bloop to create bloop config files."),
+      artifact = s""""ch.epfl.scala" % "sbt-bloop" % "$version"""",
+      resolver
     )
+  }
 
   /**
    * Short description and artifact for the sbt-metals plugin
    */
-  private def metalsPluginDetails(version: String) =
-    (
-      "This file enables semantic information to be produced by sbt.",
-      s""""org.scalameta" % "sbt-metals" % "$version""""
+  def metalsPluginDetails: PluginDetails = {
+    val resolver =
+      if (isSnapshotVersion(BuildInfo.metalsVersion))
+        Some(
+          """"Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots""""
+        )
+      else None
+
+    PluginDetails(
+      Seq(
+        "This file enables semantic information to be produced by sbt.",
+        "It also adds support for debugging using the Debug Adapter Protocol"
+      ),
+      s""""org.scalameta" % "sbt-metals" % "${BuildInfo.metalsVersion}"""",
+      resolver
+    )
+  }
+
+  /**
+   * Short description and artifact for the sbt-jdi-tools plugin
+   */
+  def jdiToolsPluginDetails: PluginDetails =
+    PluginDetails(
+      Seq(
+        "This file makes sure that the JDI tools are in the sbt classpath.",
+        "JDI tools are used by the debug adapter server."
+      ),
+      s""""org.scala-debugger" % "sbt-jdi-tools" % "1.1.1"""",
```

Comment:
Just a nit, can we actually pull this from build info so that we can get updates from Steward?

Reply:
Done!

Is Steward smart enough to find a version number in the `build.sbt` that is used in the code through build info? :thinking:
Nevertheless it's probably better to have all versions in the same place.